### PR TITLE
Add missing spec and documentation for button_tag helper

### PR DESCRIPTION
**Pull request by akshay-vishnoi**

### CODE REVIEW

File: actionview/lib/action_view/helpers/form_tag_helper.rb [hunk 1]
```diff
@@ -449,7 +449,8 @@ def submit_tag(value = "Save changes", options = {})
       # JavaScript, for example. You can use the button tag as a regular
       # submit tag but it isn't supported in legacy browsers. However,
       # the button tag allows richer labels such as images and emphasis,
```
Comment (BenMorganIO) [inline actionview/lib/action_view/helpers/form_tag_helper.rb]: Missed an optimization. `the button tag allows` doesn't read clearly. Maybe `the button tag does allow for richer labels`.

File: actionview/lib/action_view/helpers/form_tag_helper.rb [hunk 2]
```diff
@@ -449,7 +449,8 @@ def submit_tag(value = "Save changes", options = {})
...
-      # so this helper will also accept a block.
+      # so this helper will also accept a block. By default, it will create
+      # button tag with type `submit`, if type is not given.
```
Comment (BenMorganIO) [inline actionview/lib/action_view/helpers/form_tag_helper.rb]: Missing preposition. `it will create _a_ button tag`

**Comment by BenMorganIO** [comment]
:+1:

File: actionview/lib/action_view/helpers/form_tag_helper.rb (hunk 1 above)
Comment (akshay-vishnoi) [inline actionview/lib/action_view/helpers/form_tag_helper.rb]: I think it is good as it is. Lets see what others suggest.

Comment (rafaelfranca) [inline actionview/lib/action_view/helpers/form_tag_helper.rb]: :+1:

Comment (akshay-vishnoi) [inline actionview/lib/action_view/helpers/form_tag_helper.rb]: @rafaelfranca - oops.... confusing thumbs up..... do you want me to change it or it is good as it is? :smile:

**Comment by akshay-vishnoi** [comment]
@rafaelfranca, @BenMorganIO  - PR updated. Thanks for suggestions.

**Comment by arthurnn** [comment]
thanks

**Pull request merged**